### ISSUE TITLE
HNC: Delete validation for Namespace and HNS is broken on k8s versions <1.15

### ISSUE THREAD

**Issue by yiqigao217** [kubernetes-sigs/multi-tenancy#680]
See exact same issue in controller-runtime repo - [#845](https://github.com/kubernetes-sigs/controller-runtime/issues/845).

The HNC webhook validation on `kubectl delete ns <namespace>` and `kubectl delete hns <subnamespace> -n <ownerNamespace>` is broken on k8s versions <1.15.

You should get `Forbidden`:
```
Error from server (Forbidden): admission webhook "vnamespace.k8s.io" denied the request: The namespace zchild is an owned namespace. Please delete the HNS instance from the owner namespace zparent instead.
```
instead of `BadRequest`:
```
Error from server (BadRequest): admission webhook "vnamespace.k8s.io" denied the request: there is no content to decode
```

**Comment by adrianludwin** [comment]
/good-first-issue

I don't think this is very high priority for us to fix (1.14 is on its way out the door anyway, at least on GKE) but if someone wants to help out or we get a strong signal that this fix is needed in 1.14, we can revisit this.

**Comment by adrianludwin** [comment]
1.14 is now well out of support so I don't think we need to worry about this.

/remove-lifecycle rotten
/close
